libminifi/src/processors/LogAttribute.cpp

```diff
@@ -38,13 +38,23 @@ namespace apache {
 namespace nifi {
 namespace minifi {
 namespace processors {
-core::Property LogAttribute::LogLevel("Log Level", "The Log Level to use when logging the Attributes", "info");
-core::Property LogAttribute::AttributesToLog("Attributes to Log", "A comma-separated list of Attributes to Log. If not specified, all attributes will be logged.", "");
-core::Property LogAttribute::AttributesToIgnore("Attributes to Ignore", "A comma-separated list of Attributes to ignore. If not specified, no attributes will be ignored.", "");
-core::Property LogAttribute::LogPayload("Log Payload", "If true, the FlowFile's payload will be logged, in addition to its attributes;"
-                                        "otherwise, just the Attributes will be logged.",
-                                        "false");
-core::Property LogAttribute::LogPrefix("Log prefix", "Log prefix appended to the log lines. It helps to distinguish the output of multiple LogAttribute processors.", "");
+
+core::Property LogAttribute::LogLevel(
+    core::PropertyBuilder::createProperty("Log Level")->withDescription("The Log Level to use when logging the Attributes")->withAllowedValue<std::string>("info")->withAllowedValue("trace")
+        ->withAllowedValue("error")->withAllowedValue("warn")->withAllowedValue("debug")->withDefaultValue("info")->build());
```

Comment:
I don't like this, withAllowedValue should be implemented as 
```
template<typename T>
withAllowedValue(std::set<T>) {
  ...
}
```

This way it can be done as:
```
withAllowedValue({"Red", "Green", "Blue"))
```

Reply:
I think that would be "withAllowedValues". That's a good point. What you don't like is an artifact of a fluent API and is something that withAllowedValues would solve. Good idea.